### PR TITLE
From tests

### PR DESCRIPTION
**Pull request by lazedo**
* adds `anti-action` for use in `dialplan condition`
* allows correlation on the inbound leg
* fixes connecting-leg
* verifies node split in kz_dist
* conference event binding
* adds callflow modules for composition

### CODE REVIEW

**Comment by lazedo** [comment]
some more to come after this

File: applications/callflow/src/cf_exe.erl [hunk 1]
```diff
@@ -918,7 +918,7 @@ handle_channel_direct(Call, JObj, Notify) ->
     NewFetchId = kz_json:get_value([<<"Custom-Channel-Vars">>, <<"Fetch-ID">>], JObj),
     case OrgFetchId =:= NewFetchId of
         'true' ->
-            ReplacedBy = kz_json:get_value(<<"Connecting-Leg-A-UUID">>, JObj),
+            ReplacedBy = kz_json:get_value(<<"Connecting-Leg-B-UUID">>, JObj),
```
Comment (jamesaimonetti) [inline applications/callflow/src/cf_exe.erl]: You should add this to kz_call_event.erl and update this function's spec. Maybe even rename JObj to CallEvent or something descriptive :)

File: applications/callflow/src/module/cf_answer.erl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+-behaviour(gen_cf_action).
+
+-export([handle/2]).
+
+-include("callflow.hrl").
+
+-spec handle(kz_json:object(), kapps_call:call()) -> 'ok'.
+handle(_Data, Call) ->
+    lager:debug("answering call"),
```
Comment (jamesaimonetti) [inline applications/callflow/src/module/cf_answer.erl]: callflow actions should log at info so they're seen more readily.

File: applications/callflow/src/module/cf_noop.erl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+-spec handle(kz_json:object(), kapps_call:call()) -> 'ok'.
+handle(_Data, Call) ->
+    lager:debug("sending noop"),
+    NoopId = kapps_call_command:noop(Call),
+    lager:debug("noop ~s sent", [NoopId]),
+    case kapps_call_command:wait_for_noop(Call, NoopId) of
+        {'error', 'channel_hungup'} ->
+            lager:debug("noop not received"),
+            cf_exe:stop(Call);
+        {ok, _} ->
```
Comment (jamesaimonetti) [inline applications/callflow/src/module/cf_noop.erl]: Tickie and log at info

File: applications/ecallmgr/src/call_control/ecallmgr_call_control.erl [hunk 4]
```diff
@@ -1025,7 +1027,7 @@ handle_direct(JObj, #state{fetch_id=FetchId
                           }=State) ->
     case kz_call_event:custom_channel_var(JObj, <<"Fetch-ID">>) of
         FetchId ->
-            ReplacedBy = kz_json:get_ne_binary_value(<<"Connecting-Leg-A-UUID">>, JObj),
+            ReplacedBy = kz_json:get_ne_binary_value(<<"Connecting-Leg-B-UUID">>, JObj),
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/call_control/ecallmgr_call_control.erl]: Move to kz_call_event accessor

**Pull request merged**